CMakeLists.txt

```diff
@@ -644,6 +729,20 @@ if(USE_CUDA)
   target_link_libraries(_lightgbm PRIVATE ${histograms})
 endif()
 
+if(USE_ROCM)
```

Comment:
Can we merge CUDA and HIP with `if( USE_CUDA OR USE_ROCM)` here?

Reply:
Done in [8f6600e](https://github.com/microsoft/LightGBM/pull/6086/commits/8f6600edd7d5336d448c9e49b084491a10d7f501).